Segments/Blackburrow.mapproj

```diff
@@ -269807,6 +269826,193 @@ return yima;
+               case 3: var gemDrop3 = new HeartShapedRuby(2000000);
+                       gemDrop3.Move(player.Location, true, player.Segment); break;
+              }
+              
+          }
+
+]]></block>
+        <block><![CDATA[]]></block>
+      </script>
+      <script name="OnIncomingPlayer" enabled="true">
+        <block><![CDATA[]]></block>
+        <block><![CDATA[
+
+  if (engageRecently(source))
+    return;
+
+                var sendLocation = new Point2D(6,3,9);
+                var targetSegment = source.Facet.GetSegmentByIndex(1, false);
+                var listStun = source.GetBeheldInVisibility(3).OfType<PlayerGroup>().SelectMany(s => s.Members).ToList();
+                var randBody = Utility.RandomBetween(1, 248);
+                var location = source.Location;
+                var targetSegment2 = source.Facet.GetSegmentByIndex(29, false);
+                var playerBody = player.Body;
+
+
+switch (Utility.Random(1, 12))
+    {
+        case 1:  source.Say($"Oh ya want to fight do ya! Let's Have at it!"); break;
+        case 2:  source.Say($"{player.Name}, I will make you my wench!"); break;    
+        case 3: player.Fear(12); source.Say($"Hahahahaha - Dance puppet, Dance!"); break;
+        case 4: Timer.DelayCall(TimeSpan.FromSeconds(6), () =>
+                     {
+                      source.Say($"Oh forget this, {player.Name}, I'm outta here!");
+                     
+                     
+                      if (listStun != null)
+                      {
+                      foreach (var person in listStun)
+                      {
+
+                          person.Stun(10);
+
+                      }
+                      }
+
+                      source.Teleport(new Point2D(6,9,10));
+                      
+                      });
+                      
+                      break;
+                      
+        case 5: source.Say($"Zark! Goodbye!"); player.Teleport(new Point2D(40, 13, 10)); break;
+        case 6: source.Say($"Good Luck!"); player.Teleport(new Point2D(-41, 43, 10)); break;
+        case 7: source.Say("Don't let the door hit you on the way out!"); player.Teleport(new Point2D(-51, 19, 10)); break;
+        case 8: source.Say("You'll never steal my toothpaste!"); player.Teleport(new Point2D(21, -18, 10)); break;
+        case 9: source.Emote($"punches {player.Name}'s soul out of his body!");
+                player.Stun(5);
+                player.EmitSound(187,3,9);
+                player.Body = 66;
+                player.Teleport(sendLocation, targetSegment); 
+                Timer.DelayCall(TimeSpan.FromSeconds(3), () =>
+                       {
+                       if (player.IsAlive)
+                       {
+                       player.EmitSound(187,3,9);
+                       player.Teleport(location, targetSegment2);
+                       player.Body = playerBody;
```

Comment:
A little hesitant about changing a body type to a ghost (66) and then back.  - If for whatever reason the user dies prior to the timer going off, the players body will not be reset, due to the "if (player.IsAlive)". You need to reset it without this line of code regardless.